src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -3750,7 +3750,7 @@ Give the compiler some way to differentiate the methods. For example, you can gi
     <value>'{0}' does not implement '{1}'</value>
   </data>
   <data name="ERR_TaskRetNoObjectRequired" xml:space="preserve">
-    <value>Since '{0}' is an async method that returns 'Task', a return keyword must not be followed by an object expression. Did you intend to return 'Task&lt;T&gt;'?</value>
+    <value>Since '{0}' is an async method that returns void task-like type, a return keyword must not be followed by an object expression</value>
```

Comment:
I think we should use the return type of the method here and below.

Reply:
`Since '{0}' is an async method that returns '{1}'` - I don't think such message is representative enough. What if `{1}` is some custom async type with unclear name, e.g. `C`. `Since 'SomeClass.M' is an async method that returns 'C', a return keyword must not be followed by an object expression` - this is kinda confusing, don't you think so?

`Since '{0}' is an async method that returns void task-like type '{1}'` - here if `{1}` contains word `Task` in its name the message will be a bit overloaded, e.g. Since `'SomeClass.M' is an async method that returns void task-like type 'Task', a return keyword must not be followed by an object expression`

Which one do you preffer or did you mean some other variant?

Reply:
As a user I would tend to expect that if the return type is non-generic, then you can't return a value, and if it is generic, then you can return a value. I think a custom Task type where this *isn't* clear is rare enough that we don't need to account for it.